README.md

```diff
@@ -114,6 +110,45 @@ history.listen((location, action) => {
 })
 ```
 
+Post-change listener functions will be not be invoked when a transition is blocked by a pre-change hook, unless the user used the built-in browser back or forward features to initiate the transition. In this case, the listener functions will be called with the same `location` and `action` as the previous invocation.
+
+#### Pre-change hook
+You can perform side effects before history state changes by registering functions with `history.before`
+
+```js
+const unsub = history.before((location, action) => {
+  location.query = queryParseFn(location.search)
```

Comment:
This use case is legit for query parsing, but I can't think of many other use cases. This use case could be solved in a lot of other ways though, like just wrapping all your `listen` handlers before you use them.

```js
function parseQuery(callback) {
  return function (location) {
    if (!location.query)
      location.query = parseTheQueryString(location.search)

    callback(location)
  }
}

history.listen(parseQuery(callback))
```